graphql/resolvers.go

```diff
@@ -1502,37 +1502,31 @@ func (r *queryResolver) Projects(ctx context.Context) ([]*GroupedProjects, error
 		return nil, ResourceNotFound.Send(ctx, err.Error())
 	}
 
-	groupsMap := make(map[string][]*restModel.APIProjectRef)
-
-	for _, p := range allProjs {
-		groupName := strings.Join([]string{p.Owner, p.Repo}, "/")
-		apiProjectRef := restModel.APIProjectRef{}
-		if err = apiProjectRef.BuildFromService(p); err != nil {
-			return nil, InternalServerError.Send(ctx, fmt.Sprintf("error building APIProjectRef from service: %s", err.Error()))
-		}
-
-		if projs, ok := groupsMap[groupName]; ok {
-			groupsMap[groupName] = append(projs, &apiProjectRef)
-		} else {
-			groupsMap[groupName] = []*restModel.APIProjectRef{&apiProjectRef}
-		}
+	groupedProjects, err := GroupProjects(allProjs, false)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("error grouping project: %s", err.Error()))
 	}
+	return groupedProjects, nil
+}
 
-	groupsArr := []*GroupedProjects{}
+func (r *queryResolver) ViewableProjects(ctx context.Context) ([]*GroupedProjects, error) {
+	usr := MustHaveUser(ctx)
+	projectIds, err := usr.GetViewableProjects()
```

Comment:
I wonder if these names should be changed to GetViewableProjectSettings or something like that, to distinguish between "user can view tasks for this project" vs "users can view admin settings for this project"

Reply:
good point. done.